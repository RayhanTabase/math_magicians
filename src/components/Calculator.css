button {
  background: rgb(233, 233, 233);
  height: 6em;
  border: 1px solid rgb(223, 223, 223);
}

#calculatorContainer {
  width: 100%;
  min-width: 100%;
  height: 100%;
  min-height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

#calculator {
  margin-top: 5rem;
  width: 70%;
  min-width: 70%;
  font-size: 1.4rem;
}

#resultsContainer {
  box-sizing: border-box;
  max-width: 100%;
  min-width: 100%;
  background: gray;
  color: #fff;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  align-items: flex-end;
  min-height: 3em;
  padding-right: 0.5em;
}

#resultsContainer .currentCalculation {
  font-size: 0.6em;
}

#calculatorBody {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
}

.gridSpanTwo {
  grid-column: 1 / span 2;
}

.altColor {
  background: orange;
}
