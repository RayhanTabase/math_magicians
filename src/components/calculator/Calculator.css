
button {
  background: rgb(233, 233, 233);
  height: 6em;
  border: 1px solid rgb(223, 223, 223);
}

#calculatorContainer {
  display: flex;
  justify-content: space-between;
}

#calculatorContainer .header {
  font-size: 1.2rem;
  width: 20%;
}

#calculator {
  width: 60%;
}

#resultsContainer {
  box-sizing: border-box;
  max-width: 100%;
  min-width: 100%;
  background: gray;
  color: #fff;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  align-items: flex-end;
  min-height: 3em;
  padding-right: 0.5em;
}

#resultsContainer .currentCalculation {
  font-size: 0.6em;
}

#calculatorBody {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
}

.gridSpanTwo {
  grid-column: 1 / span 2;
}

.altColor {
  background: orange;
}
