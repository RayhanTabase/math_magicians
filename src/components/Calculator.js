import React, { useState } from 'react';
import './Calculator.css';

import calculate from '../logic/calculate';

function Calculator() {
  const [data, setData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const [currentCalc, setCurrentCalc] = useState({
    calculation: false,
    num1: null,
    num2: null,
    prevTotal: null,
    prevOperation: null,
  });

  const handleClick = async (e) => {
    const buttonName = e.target.name;
    const response = await calculate(data, buttonName);
    setData({
      total: response.total,
      next: response.next,
      operation: response.operation,
    });

    if (buttonName === '=') {
      if (response.total) {
        setCurrentCalc({
          calculation: true,
          num1: currentCalc.num1,
          num2: currentCalc.num2,
          prevTotal: response.total,
          prevOperation: currentCalc.prevOperation,
        });
      } else {
        setCurrentCalc({
          calculation: false,
          num1: null,
          num2: null,
          prevTotal: null,
          prevOperation: null,
        });
      }
    } else {
      setCurrentCalc({
        calculation: false,
        num1: response.total || response.next,
        num2: response.total ? response.next : null,
        total: null,
        prevOperation: response.operation,
      });
    }
  };

  const { total, next } = data;
  const {
    calculation, num1, num2, prevTotal, prevOperation,
  } = currentCalc;
  return (
    <div id="calculatorContainer">
      <div id="calculator">
        <div id="resultsContainer">
          <div className="currentCalculation">
            {num1}
            {prevOperation}
            {num2}
            {calculation ? '=' : false}
            {calculation ? prevTotal : false}
          </div>
          <div className="currentTotal">
            {next || total || 0}
          </div>
        </div>
        <div id="calculatorBody">
          <button type="button" onClick={handleClick} name="AC">AC</button>
          <button type="button" onClick={handleClick} name="+/-">+/-</button>
          <button type="button" onClick={handleClick} name="%">%</button>
          <button type="button" className="altColor" onClick={handleClick} name="รท">รท</button>
          <button type="button" name={7} onClick={handleClick}>7</button>
          <button type="button" name={8} onClick={handleClick}>8</button>
          <button type="button" name={9} onClick={handleClick}>9</button>
          <button type="button" className="altColor" onClick={handleClick} name="x">x</button>
          <button type="button" name={4} onClick={handleClick}>4</button>
          <button type="button" name={5} onClick={handleClick}>5</button>
          <button type="button" name={6} onClick={handleClick}>6</button>
          <button type="button" className="altColor" onClick={handleClick} name="-">-</button>
          <button type="button" name={1} onClick={handleClick}>1</button>
          <button type="button" name={2} onClick={handleClick}>2</button>
          <button type="button" name={3} onClick={handleClick}>3</button>
          <button type="button" className="altColor" onClick={handleClick} name="+">+</button>
          <button type="button" className="gridSpanTwo" name={0} onClick={handleClick}>0</button>
          <button type="button" onClick={handleClick} name=".">.</button>
          <button type="button" className="altColor" onClick={handleClick} name="=">=</button>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
