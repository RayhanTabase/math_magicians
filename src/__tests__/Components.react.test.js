import React from 'react';
import { render, screen, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import Calculator from '../components/calculator/Calculator';

describe('Calculator', () => {
  const arr = [
    'AC',
    '+/-',
    '+',
    '-',
    '=',
    'รท',
    'x',
    '%',
    '1',
    '2',
    '3',
    '4',
    '5',
    '6',
    '7',
    '8',
    '9',
    '0',
    '.',
  ];

  render(<Calculator />);

  test('should render calculator and its children', async () => {
    const buttons = screen.getAllByRole('button');
    await act(async () => {
      buttons.forEach((item) => {
        userEvent.click(item);
        expect(item).not.toBeDisabled();
        expect(arr).toContain(item.name);
      });
    });
    expect(typeof buttons === 'object').toEqual(true);
    expect(buttons.length).toBe(19);
  });
});
