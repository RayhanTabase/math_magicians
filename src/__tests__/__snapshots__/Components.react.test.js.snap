// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `<DocumentFragment />`;

exports[`Calculator rsnapshot test for Calculator component 1`] = `<DocumentFragment />`;

exports[`Calculator snapshot test for Calculator component 1`] = `<DocumentFragment />`;

exports[`Home  1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
      Welcome to our page !
    </h2>
    <br />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
    non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium
    doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore
    veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim
    ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia
    consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque
    porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur,
    adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et
    dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis
    nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid
    ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea
    voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem
    eum fugiat quo voluptas nulla pariatur?
    </p>
  </div>
</DocumentFragment>
`;

exports[`Home renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
      Welcome to our page !
    </h2>
    <br />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
    non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium
    doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore
    veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim
    ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia
    consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque
    porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur,
    adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et
    dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis
    nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid
    ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea
    voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem
    eum fugiat quo voluptas nulla pariatur?
    </p>
  </div>
</DocumentFragment>
`;

exports[`Home snapshot for  1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
      Welcome to our page !
    </h2>
    <br />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
    non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium
    doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore
    veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim
    ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia
    consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque
    porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur,
    adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et
    dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis
    nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid
    ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea
    voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem
    eum fugiat quo voluptas nulla pariatur?
    </p>
  </div>
</DocumentFragment>
`;

exports[`Home snapshot test for Home component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
      Welcome to our page !
    </h2>
    <br />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
    non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium
    doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore
    veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim
    ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia
    consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque
    porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur,
    adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et
    dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis
    nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid
    ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea
    voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem
    eum fugiat quo voluptas nulla pariatur?
    </p>
  </div>
</DocumentFragment>
`;

exports[`Home snapshot testu for Home component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
      Welcome to our page !
    </h2>
    <br />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
    non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium
    doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore
    veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim
    ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia
    consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque
    porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur,
    adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et
    dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis
    nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid
    ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea
    voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem
    eum fugiat quo voluptas nulla pariatur?
    </p>
  </div>
</DocumentFragment>
`;

exports[`Navbar renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <h1
      class="header"
    >
      Math Magicians
    </h1>
    <ul
      class="nav-links"
    >
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class=""
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class=""
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Navbar snapshot test for Home component 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <h1
      class="header"
    >
      Math Magicians
    </h1>
    <ul
      class="nav-links"
    >
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class=""
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class=""
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Navbar snapshot test for Navbar component 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <h1
      class="header"
    >
      Math Magicians
    </h1>
    <ul
      class="nav-links"
    >
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class=""
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class=""
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Quotes should render the quotes 1`] = `
<DocumentFragment>
  <div
    class="container quoteContainer"
  >
    <p>
      <span>
        Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.
      </span>
      <span>
          -  
      </span>
      <span
        class="italize"
      >
        William Paul Thurston
      </span>
    </p>
  </div>
</DocumentFragment>
`;
